invokeai/app/invocations/ip_adapter.py

```diff
@@ -102,19 +114,25 @@ def invoke(self, context: InvocationContext) -> IPAdapterOutput:
         )
 
     def _get_image_encoder(self, context: InvocationContext, image_encoder_model_name: str) -> AnyModelConfig:
-        found = False
-        while not found:
+        image_encoder_models = context.models.search_by_attrs(
+            name=image_encoder_model_name, base=BaseModelType.Any, type=ModelType.CLIPVision
+        )
+
+        if not len(image_encoder_models) > 0:
+            context.logger.warning(
+                f"The image encoder required by this IP Adapter ({image_encoder_model_name}) is not installed. \
+                    Downloading and installing now. This may take a while."
+            )
+
+            installer = context._services.model_manager.install
+            job = installer.heuristic_import(f"InvokeAI/{image_encoder_model_name}")
+            installer.wait_for_job(job, timeout=600)  # Wait for up to 10 minutes
             image_encoder_models = context.models.search_by_attrs(
                 name=image_encoder_model_name, base=BaseModelType.Any, type=ModelType.CLIPVision
             )
-            found = len(image_encoder_models) > 0
-            if not found:
-                context.logger.warning(
-                    f"The image encoder required by this IP Adapter ({image_encoder_model_name}) is not installed."
-                )
-                context.logger.warning("Downloading and installing now. This may take a while.")
-                installer = context._services.model_manager.install
-                job = installer.heuristic_import(f"InvokeAI/{image_encoder_model_name}")
-                installer.wait_for_job(job, timeout=600)  # wait up to 10 minutes - then raise a TimeoutException
-        assert len(image_encoder_models) == 1
+
+            if len(image_encoder_models) == 0:
+                context.logger.error("Error while fetching CLIP Vision Image Encoder")
+                assert len(image_encoder_models) == 1
```

Comment:
Ye not a big of how I did this but essentially we're saying "after the model is downloaded, if by any chance the image encoder list is still 0, then we just raise the assertion and end the function" .. because the output of the function is set to get the first item in the retrieved list which will not exist if the download fails. Probably can be done cleaner.

Also not a fan of the fact that I need to call on the `search_by_attrs` twice .. but I need to check if the `ModelInstallJob` has some kind of callback on it. Did this in a rush, so didn't find the time to check if there was one.